api/integration/app/install/config_test.go

```diff
@@ -0,0 +1,741 @@
+		require.NoError(t, err)
+
+		// Create a request with invalid token
+		req := httptest.NewRequest(http.MethodPost, s.baseURL+"/app/config/template", bytes.NewReader(reqBodyBytes))
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Authorization", "Bearer INVALID_TOKEN")
+		rec := httptest.NewRecorder()
+
+		// Serve the request
+		router.ServeHTTP(rec, req)
+
+		// Check the response
+		assert.Equal(t, http.StatusUnauthorized, rec.Code)
+		assert.Equal(t, "application/json", rec.Header().Get("Content-Type"))
+
+		// Parse the response body
+		var apiError types.APIError
+		err = json.NewDecoder(rec.Body).Decode(&apiError)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusUnauthorized, apiError.StatusCode)
+	})
+
+	// Test invalid JSON request
+	s.T().Run("Invalid JSON request", func(t *testing.T) {
+		// Create an install controller with the app config
+		apiInstance := s.createAPI(t, states.StateNew, &release.ReleaseData{
+			AppConfig: &appConfigWithTemplates,
+		}, nil)
+
+		// Create a router and register the API routes
+		router := mux.NewRouter()
+		apiInstance.RegisterRoutes(router)
+
+		// Create a request with invalid JSON
+		req := httptest.NewRequest(http.MethodPost, s.baseURL+"/app/config/template", bytes.NewReader([]byte("invalid json")))
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Authorization", "Bearer TOKEN")
+		rec := httptest.NewRecorder()
+
+		// Serve the request
+		router.ServeHTTP(rec, req)
+
+		// Check the response
+		assert.Equal(t, http.StatusBadRequest, rec.Code)
+		assert.Equal(t, "application/json", rec.Header().Get("Content-Type"))
+
+		// Parse the response body
+		var apiError types.APIError
+		err := json.NewDecoder(rec.Body).Decode(&apiError)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusBadRequest, apiError.StatusCode)
+	})
+}
+
+// Runner function that executes the suite for both install types
+func TestAppInstallSuite(t *testing.T) {
+	installTypes := []struct {
+		name        string
+		installType string
+		createAPI   func(t *testing.T, initialState statemachine.State, rc *release.ReleaseData, configValues types.AppConfigValues) *api.API
+		baseURL     string
+	}{
+		{
+			name:        "linux install",
+			installType: "linux",
+			createAPI: func(t *testing.T, initialState statemachine.State, rc *release.ReleaseData, configValues types.AppConfigValues) *api.API {
+				controller, err := linuxinstall.NewInstallController(
+					linuxinstall.WithStateMachine(linuxinstall.NewStateMachine(linuxinstall.WithCurrentState(initialState))),
+					linuxinstall.WithReleaseData(rc),
+					linuxinstall.WithConfigValues(configValues),
+				)
+				require.NoError(t, err)
+				// Create the API with the install controller
+				return integration.NewAPIWithReleaseData(t,
+					api.WithLinuxInstallController(controller),
+					api.WithAuthController(auth.NewStaticAuthController("TOKEN")),
+					api.WithLogger(logger.NewDiscardLogger()),
+				)
+			},
```

Comment:
While looking for a way to create the API instance on a per test scenario I was thinking that it would be helpful to actually consolidate the Controller options on a single shared struct for all the controllers. This way we could be passing over generic Controller options on each test that could be applied here without much fuss. Opted not to go down that route just yet since this does the trick and we avoid yet another set of large changes around the code.